application/src/main/java/org/togetherjava/tjbot/commands/moderation/report/ReportCommand.java

```diff
@@ -0,0 +1,175 @@
+ * {@link Config#getModMailChannelPattern()} ()}.
+ */
+
+public final class ReportCommand extends BotCommandAdapter implements MessageContextCommand {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCommand.class);
+    private static final String COMMAND_NAME = "report";
+    private static final String MESSAGE_INPUT = "message";
+    private static final int COOLDOWN_DURATION_VALUE = 3;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final Color AMBIENT_COLOR = Color.BLACK;
+    private final Cache<Long, Instant> authorToLastReportInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+    private final String configModMailChannelPattern;
+    private String reportedMessage;
+    private String reportedMessageUrl;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config to get the channel to forward reports to.
+     */
+    public ReportCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+
+        configModMailChannelPattern = config.getModMailChannelPattern();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        reportedMessage = event.getTarget().getContentRaw();
+        reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput.create(MESSAGE_INPUT, "Message", TextInputStyle.PARAGRAPH)
+            .setPlaceholder("Reason for reporting")
+            .setRequiredRange(10, 200)
+            .build();
+
+        Modal modal = Modal.create(generateComponentId(), "Report").addActionRow(body).build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), guildID);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+            return;
```

Comment:
u didnt test this. right now ur not responding to the user and the interaction crashes.